### DISCUSSION TITLE
Gives an error when updating How to fix?

### DISCUSSION THREAD

**Discussion by nloshka404** [firefly-iii/firefly-iii#7506] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Gives an error when updating
How to fix?

### Debug information

Debug information generated at 2023-05-15 00:00:00 Europe/Moscow for Firefly III version **6.0.10**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.10 |
| Firefly III API | 2.0.1 |
| DB version | 19 (exp. 19) |
| Docker | No |
| PHP | `8.2.5` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Moscow |
| Browser TZ | Europe/Moscow |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, daily |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 2097152 (2 MB) |
| Interface | fpm-fcgi |
| Default language | ru_RU |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-05-01 00:00:00 |
| Session end | 2023-05-31 23:59:59 |
| Session first | 2019-01-01 00:00:00 |
| User ID | 1 |
| User language | ru_RU |
| User locale | ru_RU |
| Attempt at "ru_RU.utf8" | 'ru_RU.utf8' |
| Attempt at "ru_RU.UTF-8" | 'ru_RU.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36 |
------------------------
```
Truncated from this point <----|ware\\Binder->handle()
#39 /var/www/buh.x512.ru/firefly-iii/app/Http/Middleware/Range.php(62): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\\Http\\Middleware\\Range->handle()
#41 /var/www/buh.x512.ru/firefly-iii/vendor/jc5/google2fa-laravel/src/Middleware.php(29): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#42 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): PragmaRX\\Google2FALaravel\\Middleware->handle()
#43 /var/www/buh.x512.ru/firefly-iii/app/Http/Middleware/Authenticate.php(75): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#44 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\\Http\\Middleware\\Authenticate->handle()
#45 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#46 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Laravel\\Passport\\Http\\Middleware\\CreateFreshApiToken->handle()
#47 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(59): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#48 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\AuthenticateSession->handle()
#49 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#50 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#51 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#52 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#53 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#54 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#55 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\StartSession->handle()
#56 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#57 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#58 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#59 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#60 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#61 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(797): Illuminate\\Pipeline\\Pipeline->then()
#62 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(776): Illuminate\\Routing\\Router->runRouteWithinStack()
#63 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(740): Illuminate\\Routing\\Router->runRoute()
#64 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(729): Illuminate\\Routing\\Router->dispatchToRoute()
#65 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\\Routing\\Router->dispatch()
#66 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#67 /var/www/buh.x512.ru/firefly-iii/app/Http/Middleware/InstallationId.php(52): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#68 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\\Http\\Middleware\\InstallationId->handle()
#69 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#70 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle()
#71 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#72 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#73 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#74 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#75 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#76 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#77 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#78 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#79 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#80 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#81 /var/www/buh.x512.ru/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#82 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): FireflyIII\\Http\\Middleware\\SecureHeaders->handle()
#83 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#84 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\\Pipeline\\Pipeline->then()
#85 /var/www/buh.x512.ru/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#86 /var/www/buh.x512.ru/firefly-iii/public/index.php(73): Illuminate\\Foundation\\Http\\Kernel->handle()
#87 {main}
"} 

```

### Steps to reproduce

cd firefly-iii-updated
rm -rf bootstrap/cache/*
php artisan cache:clear
php artisan migrate --seed
php artisan firefly-iii:upgrade-database

!!!

php artisan passport:install
php artisan cache:clear
cd ..

### Additional info

Now executing firefly-iii:budget-limit-periods

   Error

  Class "Spatie\Period\Period" not found

  at app/Handlers/Events/Model/BudgetLimitHandler.php:195
    191▕             return;
    192▕         }
    193▕
    194▕         // limit period in total is:
  ➜ 195▕         $limitPeriod = Period::make($start, $end, precision: Precision::DAY(), boundaries: Boundaries::EXCLUDE_NONE());
    196▕
    197▕         // from the start until the end of the budget limit, need to loop!
    198▕         $current = clone $start;
    199▕         while ($current <= $end) {

  1   app/Handlers/Events/Model/BudgetLimitHandler.php:62
      FireflyIII\Handlers\Events\Model\BudgetLimitHandler::updateAvailableBudget()
      +6 vendor frames

  8   app/Console/Commands/Upgrade/AppendBudgetLimitPeriods.php:96
      Illuminate\Database\Eloquent\Model::save()

**Comment by JC5** [comment]
What happens when you run `composer install` ?

**Comment by nloshka404** [comment]
I need to update and not do a fresh install.
If you start a new installation, the installation is successful, but the data of the past is lost.
So far I have returned the data from the backup copy of the previous version.
It is not yet clear how to fix the error and perform an update.

**Comment by nloshka404** [comment]
# composer install
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 48

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 48
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 56

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 56
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]?

**Comment by nloshka404** [comment]
Continue as root/super user [yes]?
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 881

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 881
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 886

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 886
> @php -r "if (!(getenv('DYNO'))===false){file_exists('.env') || copy('.deploy/heroku/.env.heroku', '.env');copy('.deploy/heroku/.locales', '.locales');}"
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 44 installs, 24 updates, 0 removals
  - Downloading phpstan/phpdoc-parser (1.20.3)
  - Downloading phpdocumentor/type-resolver (1.7.1)
  - Downloading symfony/error-handler (v6.2.9)
  - Downloading symfony/http-kernel (v6.2.9)
  - Downloading brick/math (0.11.0)
  - Downloading ramsey/uuid (4.7.4)
  - Downloading league/flysystem (3.14.0)
  - Downloading laravel/framework (v10.9.0)
  - Downloading doctrine/dbal (3.6.2)
  - Downloading diglactic/laravel-breadcrumbs (v8.1.1)
  - Downloading phpstan/phpstan (1.10.14)
  - Downloading psr/http-factory (1.0.2)
  - Downloading guzzlehttp/psr7 (2.5.0)
  - Downloading symfony/psr-http-message-bridge (v2.2.0)
  - Downloading php-http/message-factory (1.1.0)
  - Downloading nyholm/psr7 (1.7.0)
  - Downloading laravel/passport (v11.8.6)
  - Downloading laravel/sanctum (v3.2.4)
  - Downloading psr/http-client (1.0.2)
  - Downloading guzzlehttp/guzzle (7.5.1)
  - Downloading laravelcollective/html (v6.4.1)
  - Downloading filp/whoops (2.15.2)
  - Downloading nunomaduro/collision (v7.5.2)
  - Downloading nunomaduro/larastan (v2.6.0)
  - Downloading phpstan/phpstan-deprecation-rules (1.1.3)
  - Downloading phpstan/phpstan-strict-rules (1.5.1)
  - Downloading sebastian/environment (6.0.1)
  - Downloading sebastian/diff (5.0.1)
  - Downloading phpunit/php-code-coverage (10.1.1)
  - Downloading phpunit/phpunit (10.1.2)
  - Downloading spatie/flare-client-php (1.3.6)
  - Downloading spatie/ignition (1.6.0)
  - Downloading spatie/laravel-ignition (2.1.0)
  - Downloading spatie/period (2.4.0)
  - Downloading symfony/http-client (v6.2.9)
  - Installing phpstan/phpdoc-parser (1.20.3): Extracting archive
  - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
  - Installing phpdocumentor/type-resolver (1.7.1): Extracting archive
  - Installing nikic/php-parser (v4.15.4): Extracting archive
  - Upgrading symfony/error-handler (v6.2.7 => v6.2.9): Extracting archive
  - Upgrading symfony/http-kernel (v6.2.8 => v6.2.9): Extracting archive
  - Upgrading brick/math (0.10.2 => 0.11.0): Extracting archive
  - Upgrading ramsey/uuid (4.7.3 => 4.7.4): Extracting archive
  - Upgrading league/flysystem (3.12.3 => 3.14.0): Extracting archive
  - Upgrading laravel/framework (v10.6.2 => v10.9.0): Extracting archive
  - Upgrading doctrine/dbal (3.6.1 => 3.6.2): Extracting archive
  - Installing composer/pcre (3.1.0): Extracting archive
  - Installing composer/class-map-generator (1.0.0): Extracting archive
  - Installing barryvdh/reflection-docblock (v2.1.0): Extracting archive
  - Installing barryvdh/laravel-ide-helper (v2.13.0): Extracting archive
  - Upgrading diglactic/laravel-breadcrumbs (v8.1.0 => v8.1.1): Extracting archive
  - Installing phpstan/phpstan (1.10.14): Extracting archive
  - Installing ergebnis/phpstan-rules (1.0.0): Extracting archive
  - Installing fakerphp/faker (v1.21.0): Extracting archive
  - Upgrading psr/http-factory (1.0.1 => 1.0.2): Extracting archive
  - Upgrading guzzlehttp/psr7 (2.4.4 => 2.5.0): Extracting archive
  - Upgrading symfony/psr-http-message-bridge (v2.1.4 => v2.2.0): Extracting archive
  - Upgrading php-http/message-factory (v1.0.2 => 1.1.0): Extracting archive
  - Upgrading nyholm/psr7 (1.5.1 => 1.7.0): Extracting archive
  - Upgrading laravel/passport (v11.8.5 => v11.8.6): Extracting archive
  - Upgrading laravel/sanctum (v3.2.1 => v3.2.4): Extracting archive
  - Upgrading psr/http-client (1.0.1 => 1.0.2): Extracting archive
  - Upgrading guzzlehttp/guzzle (7.5.0 => 7.5.1): Extracting archive
  - Upgrading laravelcollective/html (v6.4.0 => v6.4.1): Extracting archive
  - Installing hamcrest/hamcrest-php (v2.0.1): Extracting archive
  - Installing mockery/mockery (1.5.1): Extracting archive
  - Upgrading filp/whoops (2.15.1 => 2.15.2): Extracting archive
  - Upgrading nunomaduro/collision (v7.4.0 => v7.5.2): Extracting archive
  - Installing phpmyadmin/sql-parser (5.7.0): Extracting archive
  - Installing nunomaduro/larastan (v2.6.0): Extracting archive
  - Installing phpstan/phpstan-deprecation-rules (1.1.3): Extracting archive
  - Installing phpstan/phpstan-strict-rules (1.5.1): Extracting archive
  - Installing sebastian/version (4.0.1): Extracting archive
  - Installing sebastian/type (4.0.0): Extracting archive
  - Installing sebastian/recursion-context (5.0.0): Extracting archive
  - Installing sebastian/object-reflector (3.0.0): Extracting archive
  - Installing sebastian/object-enumerator (5.0.0): Extracting archive
  - Installing sebastian/global-state (6.0.0): Extracting archive
  - Installing sebastian/exporter (5.0.0): Extracting archive
  - Installing sebastian/environment (6.0.1): Extracting archive
  - Installing sebastian/diff (5.0.1): Extracting archive
  - Installing sebastian/comparator (5.0.0): Extracting archive
  - Installing sebastian/code-unit (2.0.0): Extracting archive
  - Installing sebastian/cli-parser (2.0.0): Extracting archive
  - Installing phpunit/php-timer (6.0.0): Extracting archive
  - Installing phpunit/php-text-template (3.0.0): Extracting archive
  - Installing phpunit/php-invoker (4.0.0): Extracting archive
  - Installing phpunit/php-file-iterator (4.0.1): Extracting archive
  - Installing theseer/tokenizer (1.2.1): Extracting archive
  - Installing sebastian/lines-of-code (2.0.0): Extracting archive
  - Installing sebastian/complexity (3.0.0): Extracting archive
  - Installing sebastian/code-unit-reverse-lookup (3.0.0): Extracting archive
  - Installing phpunit/php-code-coverage (10.1.1): Extracting archive
  - Installing phar-io/version (3.2.1): Extracting archive
  - Installing phar-io/manifest (2.0.3): Extracting archive
  - Installing myclabs/deep-copy (1.11.1): Extracting archive
  - Installing phpunit/phpunit (10.1.2): Extracting archive
  - Upgrading spatie/flare-client-php (1.3.5 => 1.3.6): Extracting archive
  - Upgrading spatie/ignition (1.4.5 => 1.6.0): Extracting archive
  - Upgrading spatie/laravel-ignition (2.0.0 => 2.1.0): Extracting archive
  - Installing spatie/period (2.4.0): Extracting archive
  - Upgrading symfony/http-client (v6.2.8 => v6.2.9): Extracting archive
  - Installing thecodingmachine/phpstan-strict-rules (v1.0.0): Extracting archive
Package laravelcollective/html is abandoned, you should avoid using it. Use spatie/laravel-html instead.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
103 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> @php artisan firefly:instructions install
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v6.0.9!                                |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
> @php artisan firefly-iii:verify-security-alerts

**Comment by nloshka404** [comment]
]# composer install
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 48

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 48
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 56

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/vendor/symfony/console/Command/DumpCompletionCommand.php on line 56
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]?
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 881

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 881
PHP Deprecated:  Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 886

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/local/bin/composer/src/Composer/Autoload/AutoloadGenerator.php on line 886
> @php -r "if (!(getenv('DYNO'))===false){file_exists('.env') || copy('.deploy/heroku/.env.heroku', '.env');copy('.deploy/heroku/.locales', '.locales');}"
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 1 install, 24 updates, 1 removal
  - Downloading phpstan/phpdoc-parser (1.20.4)
  - Downloading symfony/var-dumper (v6.2.10)
  - Downloading symfony/process (v6.2.10)
  - Downloading symfony/mime (v6.2.10)
  - Downloading symfony/http-foundation (v6.2.10)
  - Downloading symfony/error-handler (v6.2.10)
  - Downloading symfony/http-kernel (v6.2.10)
  - Downloading symfony/console (v6.2.10)
  - Downloading league/flysystem (3.15.1)
  - Downloading league/flysystem-local (3.15.0)
  - Downloading laravel/framework (v10.10.1)
  - Downloading phpstan/phpstan (1.10.15)
  - Downloading ergebnis/phpstan-rules (2.0.0)
  - Downloading nyholm/psr7 (1.8.0)
  - Downloading firebase/php-jwt (v6.5.0)
  - Downloading laravel/passport (v11.8.7)
  - Downloading laravel/sanctum (v3.2.5)
  - Downloading sebastian/diff (5.0.3)
  - Downloading phpunit/php-file-iterator (4.0.2)
  - Downloading phpunit/phpunit (10.1.3)
  - Downloading twig/twig (v3.6.0)
  - Downloading spatie/ignition (1.7.0)
  - Downloading spatie/laravel-ignition (2.1.2)
  - Downloading symfony/http-client (v6.2.10)
  - Downloading symfony/mailgun-mailer (v6.2.10)
  - Removing php-http/message-factory (1.1.0)
  - Upgrading phpstan/phpdoc-parser (1.20.3 => 1.20.4): Extracting archive
  - Upgrading symfony/var-dumper (v6.2.8 => v6.2.10): Extracting archive
  - Upgrading symfony/process (v6.2.8 => v6.2.10): Extracting archive
  - Upgrading symfony/mime (v6.2.7 => v6.2.10): Extracting archive
  - Upgrading symfony/http-foundation (v6.2.8 => v6.2.10): Extracting archive
  - Upgrading symfony/error-handler (v6.2.9 => v6.2.10): Extracting archive
  - Upgrading symfony/http-kernel (v6.2.9 => v6.2.10): Extracting archive
  - Upgrading symfony/console (v6.2.8 => v6.2.10): Extracting archive
  - Upgrading league/flysystem (3.14.0 => 3.15.1): Extracting archive
  - Installing league/flysystem-local (3.15.0): Extracting archive
  - Upgrading laravel/framework (v10.9.0 => v10.10.1): Extracting archive
  - Upgrading phpstan/phpstan (1.10.14 => 1.10.15): Extracting archive
  - Upgrading ergebnis/phpstan-rules (1.0.0 => 2.0.0): Extracting archive
  - Upgrading nyholm/psr7 (1.7.0 => 1.8.0): Extracting archive
  - Upgrading firebase/php-jwt (v6.4.0 => v6.5.0): Extracting archive
  - Upgrading laravel/passport (v11.8.6 => v11.8.7): Extracting archive
  - Upgrading laravel/sanctum (v3.2.4 => v3.2.5): Extracting archive
  - Upgrading sebastian/diff (5.0.1 => 5.0.3): Extracting archive
  - Upgrading phpunit/php-file-iterator (4.0.1 => 4.0.2): Extracting archive
  - Upgrading phpunit/phpunit (10.1.2 => 10.1.3): Extracting archive
  - Upgrading twig/twig (v3.5.1 => v3.6.0): Extracting archive
  - Upgrading spatie/ignition (1.6.0 => 1.7.0): Extracting archive
  - Upgrading spatie/laravel-ignition (2.1.0 => 2.1.2): Extracting archive
  - Upgrading symfony/http-client (v6.2.9 => v6.2.10): Extracting archive
  - Upgrading symfony/mailgun-mailer (v6.2.7 => v6.2.10): Extracting archive
Package laravelcollective/html is abandoned, you should avoid using it. Use spatie/laravel-html instead.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
103 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> @php artisan firefly:instructions install
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v6.0.10!                               |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
> @php artisan firefly-iii:verify-security-alerts

**Comment by nloshka404** [comment]
FireflyIII
installation and upgrade
Please wait...

 Command failed! See log files :(
Developed by James Cole, the source code is licensed under the [AGPL-3.0-or-later](https://www.gnu.org/licenses/agpl-3.0.html).

**Comment by nloshka404** [comment]
From the 4th time, the updates, oddly enough, were completed.
Why do such errors occur?
and why you need to run update commands several times.
It is very strange why the work of the script leads to different conclusions.

**Comment by JC5** [comment]
You need to make sure composer is the latest version.

**Comment by nloshka404** [comment]
Composer version 2.3.5 2022-04-13 16:43:00

**Comment by nloshka404** [comment]
update composer to 
Composer version 2.5.5 2023-03-21 11:50:05
